table/tables/partition.go

```diff
@@ -180,29 +223,48 @@ func partitionRecordKey(pid int64, handle int64) kv.Key {
 func (t *partitionedTable) locatePartition(ctx sessionctx.Context, pi *model.PartitionInfo, r []types.Datum) (int64, error) {
```

Comment:
Would you please add more test cases for locatePartition, especially the hash cases.